### PR TITLE
User authentication week 2

### PR DESCRIPTION
**Pull request by marcebd**
## Description

Fixed react lifecycles errors, created data models for user profile. After registering with email and password, users input all their profile information. The form validates errors (will have to look over them to see which ones I can make constant variables or put in another file to just import them).

## Milestones
Working user login and authentication. Login and register pages linked. Profile page to input information about each doctor, all information is connected to a specific username, only one profile can be created by account

### CODE REVIEW

File: Frontend/src/User-Authentication/Profile.jsx [hunk 1]
```diff
@@ -112,13 +114,52 @@ function Profile() {
 
     const handleSubmit = async (event) => {
         event.preventDefault();
-        const isValid = validateForm();
-        if (!isValid) {
+        console.log("Profile Data Frontend", profileData);
```
Comment (vseth1027) [inline Frontend/src/User-Authentication/Profile.jsx]: You should not have console.log() calls in production code

File: Frontend/src/User-Authentication/Profile.jsx [hunk 2]
```diff
@@ -1,112 +1,217 @@
...
+        profilePicture: ''
+    });
+
+    const handleChange = (event) => {
+        const { name, type, value, files } = event.target;
+        if (type === 'file') {
+            setProfileData(prevState => ({
+                ...prevState,
+                [name]: files[0]
+            }));
```
Comment (vseth1027) [inline Frontend/src/User-Authentication/Profile.jsx]: Nice use of the spread operator!

File: Frontend/src/User-Authentication/Profile.jsx [hunk 3]
```diff
@@ -1,112 +1,217 @@
...
+        if (!profileData.gender) newErrors.gender = 'Gender is required';
+        if (!profileData.contactNumber) newErrors.contactNumber = 'Contact number is required';
+        if (!profileData.biography) newErrors.biography = 'Biography is required';
+
+        // Validate arrays
+        if (profileData.languages.some(lang => !lang)) newErrors.languages = 'All language fields must be filled';
+        if (profileData.locations.some(loc => !loc)) newErrors.locations = 'All location fields must be filled';
+        if (profileData.education.some(edu => !edu.institution || !edu.degree || !edu.graduation)) {
+            newErrors.education = 'All fields in education must be filled';
+        }
```
Comment (vseth1027) [inline Frontend/src/User-Authentication/Profile.jsx]: Your validation checks are check for null values, which is a good start. However, you should also validate the data. For example, what happens if someone enters "abc" for their contact number?

File: Frontend/src/User-Authentication/Profile.jsx [hunk 4]
```diff
@@ -1,112 +1,217 @@
...
+
+        setErrors(newErrors);
+        return Object.keys(newErrors).length === 0;
+    };
+
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+        const isValid = validateForm();
+        if (!isValid) {
+            console.log('Validation errors:', errors);
```
Comment (vseth1027) [inline Frontend/src/User-Authentication/Profile.jsx]: Same thing, you MUST remove console.log from production code

File: Backend/server.js [hunk 5]
```diff
@@ -9,7 +9,8 @@ require("dotenv").config();
 const app = express();
 
 const PORT = process.env.PORT || 3000;
-
+const minPasswordLenght = 6;
```
Comment (vseth1027) [inline Backend/server.js]: nit: spelling is minPasswordLength

**Review by vseth1027 — APPROVED** [review]
Overall you're off to a solid start! I've commented on some areas of improvement so please look through those.

File: Backend/server.js (hunk 5 above)
Comment (marcebd) [reply Backend/server.js]: Resolved

File: Frontend/src/User-Authentication/Profile.jsx (hunk 4 above)
Comment (marcebd) [reply Frontend/src/User-Authentication/Profile.jsx]: Removed all console logs

**Pull request merged**